### PR TITLE
[1729] Add Rollover documentation

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/2wYz16l4/1729-document-rollover-steps

### Changes proposed in this pull request

- Adds a new `/docs` folder
- Adds documentation for Rollover steps

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [ ] Tested by running locally **N/A**
- [ ] Product Review **N/A**

### CODE REVIEW

**Comment by AbigailMcP** [comment]
[Document Rollover steps](https://trello.com/c/2wYz16l4/1729-document-rollover-steps)

File: docs/rollover.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Rollover
+
+Each year we close the current cycles courses and open the new cycle in a
+process we call Rollover.
```
Comment (defong) [inline docs/rollover.md:4]:
```suggestion
process we call 'Rollover'.
```

File: docs/rollover.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+This involves copying existing providers and courses to new records to allow the
+providers to update any details and then switching the API over to a new
+'recruitment cycle' which in turn releases the new courses on Find & Apply.
+
+In addition to this, providers also request places for courses that have numbers
+of applicants limited as they are generally over supplied.
+This process is called Allocations.
```
Comment (defong) [inline docs/rollover.md:12]:
```suggestion
This process is called 'Allocations'.
```

Comment (defong) [reply docs/rollover.md:12]: I'm not sure is this really rollover or a different key point altogether.

Comment (AbigailMcP) [reply docs/rollover.md:12]: Good point - I'm working on the allocations documentation now, so will move it into there 👍

File: docs/rollover.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Rollover
+
+Each year we close the current cycles courses and open the new cycle in a
+process we call 'Rollover'.
+
+This involves copying existing providers and courses to new records to allow the
+providers to update any details and then switching the API over to a new
```
Comment (dankmitchell) [inline docs/rollover.md:7]: Should we mention why we copy? i.e. the current cycle is still open for applications so we don't want to make any changes to those courses..

**Review by dankmitchell — COMMENTED** [review]
I guess we should test these steps then iterate...

File: docs/rollover.md (hunk 3 above)
Comment (AbigailMcP) [reply docs/rollover.md:7]: good idea, will add explanation

**Comment by AbigailMcP** [comment]
> I guess we should test these steps then iterate...

That's the plan!

File: docs/rollover.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# Rollover
+
+Each year we close the current cycle's courses and open the new cycle in a
+process we call 'Rollover'.
+
+This involves copying existing providers and courses to new records to allow the
+providers to update any details and then switching the API over to a new
+'recruitment cycle' which in turn releases the new courses on Find & Apply. We
```
Comment (defong) [inline docs/rollover.md:8]:
```suggestion
'recruitment cycle' which in turn releases the new courses on Find & Apply, when the courses is `published`. We
```

Comment (AbigailMcP) [reply docs/rollover.md:8]: updated!

**Review by defong — APPROVED** [review]
More or less about right

**Pull request merged**